resources/2024-06-support-secretstores-env.md

```diff
@@ -0,0 +1,417 @@
+# Support secret stores in environment variables
+
+* **Author**: Nick Beenham (@superbeeny)
+
+## Overview
+
+Currently the only way to provide kubernetes secrets to a container is to mount them through a secret volume. This is not ideal for many use cases, especially when the secret is only needed in the environment variables of the container. This proposal aims to add support for secret stores to the environment variables of a container.
```

Comment:
To 👍 this ... one of our goals with Radius is to support users existing applications. **If the user wrote code to read the database password through an env-var, they made that decision before Radius was in the picture, so that should be possible.** 

Right now it's not secure when you do this, which is what the proposal adds.